python/python/glide/async_commands/core.py

```diff
@@ -3266,6 +3266,111 @@ async def xautoclaim_just_id(
             await self._execute_command(RequestType.XAutoClaim, args),
         )
 
+    async def xinfo_groups(
+        self,
+        key: str,
+    ) -> List[Mapping[str, Union[str, int, None]]]:
+        """
+        Returns the list of all consumer groups and their attributes for the stream stored at `key`.
+
+        See https://valkey.io/commands/xinfo-groups for more details.
+
+        Args:
+            key (str): The key of the stream.
+
+        Returns:
+            List[Mapping[str, Union[str, int, None]]]: A list of mappings, where each mapping represents the attributes
```

Comment:
When it returns `None`? What if you have no groups or empty stream?

Reply:
I've documented when None is returned under the lag arg: 
`lag: the number of entries in the stream that are still waiting to be delivered to the group's consumers, or `None` when that number can't be determined.`

In the test "lag" is None if you use XGROUP SETID without specifying the entries-read arg and then call XINFO GROUPS after.

If you don't have any groups the command will return [], I covered this scenario in the tests